lib/pure/math.nim

```diff
@@ -153,6 +156,43 @@ func isNaN*(x: SomeFloat): bool {.inline, since: (1,5,1).} =
     when defined(js): fn()
     else: result = c_isnan(x)
 
+func copySign*[T: SomeFloat](x, y: T): T {.inline, since: (1, 5, 1).} =
+  ## Returns a value with the magnitude of `x` and the sign of `y`.
```

Comment:
add:
`; this works even if x or y are NaN or zero, both of which can carry a sign.`